src/cloud.c

```diff
@@ -61,40 +61,62 @@ struct cloud_callbacks {
 
 static struct cloud_callbacks cloud_cbs;
 
+static const char *get_id_from_json_obj(const json_object *jso)
+{
+	json_object *id_json;
+
+	if (!json_object_object_get_ex(jso, "id", &id_json))
+		return NULL;
+
+	if (json_object_get_type(id_json) != json_type_string)
+		return NULL;
+
+	return json_object_get_string(id_json);
+}
+
+/**
+ * Callback function to parse received message from AMQP queue and call the
+ * respective user callback function
+ *
+ * Returns true to discard the message envelope when the received message
+ * was consumed by the user or it is a malformed message. Or returns false to
+ * requeue the message envelope when the user can not consume the message.
+ */
 static bool cloud_receive_message(const char *exchange,
 				  const char *routing_key,
 				  const char *body, void *user_data)
 {
 	json_object *jso;
-	json_object *id_json;
 	struct l_queue *list;
 	const char *id;
+	bool discard = true;
```

Comment:
Change the name of this variable, I suggest **consumed** instead of **discard**.